From d06087111ae583a546615161adbdf5a9c425f3a2 Mon Sep 17 00:00:00 2001
From: Juliana Sims <jsims@igalia.com>
Date: Tue, 4 Oct 2022 09:05:58 -0700
Subject: [PATCH] [JSC] Fix assertion failures on riscv64
 https://bugs.webkit.org/show_bug.cgi?id=246025
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reviewed by Žan Doberšek.

* Source/WTF/wtf/PlatformCPU.h: Enable WTF_CPU_NEEDS_ALIGNED_ACCESS for riscv64

Canonical link: https://commits.webkit.org/255126@main
---
 Source/WTF/wtf/PlatformCPU.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WTF/wtf/PlatformCPU.h b/Source/WTF/wtf/PlatformCPU.h
index 58cf04f4d788..fd4dad9aeeca 100644
--- a/Source/WTF/wtf/PlatformCPU.h
+++ b/Source/WTF/wtf/PlatformCPU.h
@@ -295,7 +295,7 @@
 #define WTF_CPU_UNKNOWN 1
 #endif
 
-#if CPU(ARM) || CPU(MIPS) || CPU(UNKNOWN)
+#if CPU(ARM) || CPU(MIPS) || CPU(RISCV64) || CPU(UNKNOWN)
 #define WTF_CPU_NEEDS_ALIGNED_ACCESS 1
 #endif
 
